SSLHandshakeException occured when execute start-domain after enable-secure-admin

## Environment Details
* GlassFish Version (and build number): Eclipse GlassFish  6.2.1  (build master-b577-g50d68a3 2021-09-01T02:07:52+0000)
* JDK version: OpenJDK 64-Bit Server VM Temurin-11.0.12+7
* OS: CentOS Linux release 7.9.2009 (Core)
* Database:

----------

## Problem Description
"javax.net.ssl.SSLHandshakeException: No appropriate protocol" is printed in the server.log when execute start-domain after enable-secure-admin.

``` 
[2021-10-06T09:14:21.071+0900] [glassfish 6.2] [SEVERE] [] [] [tid: _ThreadID=56 _ThreadName=Thread-8] [timeMillis: 1633479261071] [levelValue: 1000] [[
  java.io.IOException: Cannot bind to URL [rmi://9d1918b9d298:8686/jmxrmi]: javax.naming.CommunicationException [Root exception is java.rmi.ConnectIOException: error during JRMP connection establishment; nested exception is:
        javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)]
        at java.management.rmi/javax.management.remote.rmi.RMIConnectorServer.newIOException(RMIConnectorServer.java:854)
        at java.management.rmi/javax.management.remote.rmi.RMIConnectorServer.start(RMIConnectorServer.java:501)
        at org.glassfish.admin.mbeanserver.RMIConnectorStarter.start(RMIConnectorStarter.java:296)
        at org.glassfish.admin.mbeanserver.JMXStartupService$JMXConnectorsStarterThread.startConnector(JMXStartupService.java:290)
        at org.glassfish.admin.mbeanserver.JMXStartupService$JMXConnectorsStarterThread.run(JMXStartupService.java:322)
Caused by: javax.naming.CommunicationException [Root exception is java.rmi.ConnectIOException: error during JRMP connection establishment; nested exception is:
        javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)]
        at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.rebind(RegistryContext.java:178)
        at java.naming/com.sun.jndi.toolkit.url.GenericURLContext.rebind(GenericURLContext.java:253)
        at java.naming/javax.naming.InitialContext.rebind(InitialContext.java:425)
        at java.naming/javax.naming.InitialContext.rebind(InitialContext.java:425)
        at java.management.rmi/javax.management.remote.rmi.RMIConnectorServer.bind(RMIConnectorServer.java:711)
        at java.management.rmi/javax.management.remote.rmi.RMIConnectorServer.start(RMIConnectorServer.java:496)
        ... 3 more
Caused by: java.rmi.ConnectIOException: error during JRMP connection establishment; nested exception is:
        javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:300)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:196)
        at java.rmi/sun.rmi.server.UnicastRef.newCall(UnicastRef.java:343)
        at java.rmi/sun.rmi.registry.RegistryImpl_Stub.rebind(RegistryImpl_Stub.java:150)
        at jdk.naming.rmi/com.sun.jndi.rmi.registry.RegistryContext.rebind(RegistryContext.java:176)
        ... 8 more
Caused by: javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
        at java.base/sun.security.ssl.HandshakeContext.<init>(HandshakeContext.java:170)
        at java.base/sun.security.ssl.ClientHandshakeContext.<init>(ClientHandshakeContext.java:98)
        at java.base/sun.security.ssl.TransportContext.kickstart(TransportContext.java:221)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:443)
        at java.base/sun.security.ssl.SSLSocketImpl.ensureNegotiated(SSLSocketImpl.java:841)
        at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1211)
        at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)
        at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:142)
        at java.base/java.io.DataOutputStream.flush(DataOutputStream.java:123)
        at java.rmi/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:222)
        ... 12 more
]]
```

* It seems to be caused by #23531

## Steps to reproduce
1. start-domain
2. change-admin-password and enable-secure-admin
3. restart-domain
4. Check the server.log

## Impact of issue
JMX connection fails.

--- a/nucleus/common/mbeanserver/src/main/java/org/glassfish/admin/mbeanserver/ssl/SSLClientConfigurator.java
+++ b/nucleus/common/mbeanserver/src/main/java/org/glassfish/admin/mbeanserver/ssl/SSLClientConfigurator.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 1997, 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Public License v. 2.0, which is available at
@@ -401,6 +402,15 @@ public class SSLClientConfigurator {
         }
         if (sslParams.getTlsEnabled()) {
             tmpSSLArtifactsList.add("TLSv1");
+        }
+        if (sslParams.getTls11Enabled()) {
+            tmpSSLArtifactsList.add("TLSv1.1");
+        }
+        if (sslParams.getTls12Enabled()) {
+            tmpSSLArtifactsList.add("TLSv1.2");
+        }
+        if (sslParams.getTls13Enabled()) {
+            tmpSSLArtifactsList.add("TLSv1.3");
         }
         if (sslParams.getSsl3Enabled() || sslParams.getTlsEnabled()) {
             tmpSSLArtifactsList.add("SSLv2Hello");

--- a/nucleus/common/mbeanserver/src/main/java/org/glassfish/admin/mbeanserver/ssl/SSLParams.java
+++ b/nucleus/common/mbeanserver/src/main/java/org/glassfish/admin/mbeanserver/ssl/SSLParams.java
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 1997, 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Public License v. 2.0, which is available at
@@ -61,6 +62,9 @@ public class SSLParams {
     private Boolean ssl3Enabled = true;
     private String ssl3TlsCiphers;
     private Boolean tlsEnabled=true;
+    private Boolean tls11Enabled=true;
+    private Boolean tls12Enabled=true;
+    private Boolean tls13Enabled=true;
     private Boolean tlsRollBackEnabled=false;
 
 
@@ -295,10 +299,33 @@ public class SSLParams {
         return tlsEnabled;
     }
 
+    public Boolean getTls11Enabled() {
+        return tls11Enabled;
+    }
+
+    public Boolean getTls12Enabled() {
+        return tls12Enabled;
+    }
+
+    public Boolean getTls13Enabled() {
+        return tls13Enabled;
+    }
+
     public void setTlsEnabled(String tlsEnabled) {
         this.tlsEnabled = Boolean.parseBoolean(tlsEnabled);
     }
 
+    public void setTls11Enabled(String tls11Enabled) {
+        this.tls11Enabled = Boolean.parseBoolean(tls11Enabled);
+    }
+
+    public void setTls12Enabled(String tls12Enabled) {
+        this.tls12Enabled = Boolean.parseBoolean(tls12Enabled);
+    }
+
+    public void setTls13Enabled(String tls13Enabled) {
+        this.tls13Enabled = Boolean.parseBoolean(tls13Enabled);
+    }
 
     /**
      * Determines whether TLS rollback is enabled. TLS rollback should be enabled for Microsoft Internet Explorer 5.0

